php_phongo.c

```diff
@@ -775,7 +813,25 @@ int phongo_execute_command(mongoc_client_t *client, const char *db, zval *zcomma
 	/* Although "opts" already always includes the serverId option, the read
 	 * preference is added to the command parts, which is relevant for mongos
 	 * command construction. */
-	if (!mongoc_client_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error)) {
+	switch (type) {
+		case PHONGO_COMMAND_RAW:
+			result = mongoc_client_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error);
+			break;
+		case PHONGO_COMMAND_READ:
+			result = mongoc_client_read_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error);
+			break;
+		case PHONGO_COMMAND_WRITE:
+			result = mongoc_client_write_command_with_opts(client, db, command->bson, opts, &reply, &error);
+			break;
+		case PHONGO_COMMAND_READ_WRITE:
+			/* We can pass NULL as readPreference, as this argument was added historically, but has no function */
+			result = mongoc_client_read_write_command_with_opts(client, db, command->bson, NULL, opts, &reply, &error);
+			break;
+		default:
+			/* Should never happen, but if it does: exception */
+			phongo_throw_exception(PHONGO_ERROR_LOGIC TSRMLS_CC, "Type '%d' should never have been passed to phongo_execute_command, please file a bug report", type);
```

Comment:
Should we return after throwing? Otherwise we might enter the `(!result)` condition below and attempt to throw again from an unset `bson_error_t`.

Reply:
done